lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -471,6 +484,88 @@ fileModeSpec =  do
             destroyDBLayer ctx
             testOpeningCleaning f (`readCheckpoint'` testPk) (Just testCp) Nothing
 
+        describe "Golden rollback scenarios" $ do
+            let dummyHash x = Hash $ x <> BS.pack (replicate (32 - (BS.length x)) 0)
+            let dummyAddr x = Address $ x <> BS.pack (replicate (32 - (BS.length x)) 0)
+
+            it "(Regression test #1575) - TxMetas and checkpoints should \
+               \rollback to the same place" $ \f -> do
+                (_ctx, db@DBLayer{..}) <- newDBLayer' (Just f)
+
+                let ourAddrs = knownAddresses (getState testCp)
+
+                atomically $ do
+                    unsafeRunExceptT $ initializeWallet testPk testCp testMetadata mempty
+
+                let mockApply h mockTxs = do
+                        Just cpA <- atomically $ readCheckpoint testPk
+                        let slotA = slotId $ currentTip cpA
+                        let Quantity bhA = blockHeight $ currentTip cpA
+                        let hashA = headerHash $ currentTip cpA
+                        let fakeBlock = Block
+                                (BlockHeader
+                                { slotId = slotA { epochNumber = (epochNumber slotA) + 1}
+                                -- Increment blockHeight by steps greater than k
+                                -- Such that old checkpoints are always pruned.
+                                , blockHeight = Quantity $ bhA + 5000
```

Comment:
I am not sure what the pruning has to do with this test and why it matters to introduce this extra complexity in the test?

Reply:
Hm… indeed

Reply:
No, this is essential. Without pruning there is no problem.

If a checkpoint exists for every slot, then `nearestPoint` and `requestedPoint` are going to be equal, and the difference does not matter (Also see diagram in #1575)

Reply:
But you don't have to insert a checkpoint at every slot for the test in the first place?

Reply:
I wanted to include [`prune`](https://github.com/input-output-hk/cardano-wallet/blob/7e0aa90186462721397b22a3ca90967862716fbb/lib/core/src/Cardano/Wallet.hs#L767) similar to `restoreBlocks`, to convince myself that the bug could happen in reality.

But, yes, we could also specify which checkpoints to write manually.